src/pal/pal_apple.h

```diff
@@ -50,6 +52,40 @@ namespace snmalloc
       else
         ::bzero(p, size);
     }
+
+    /**
+     * Overriding here to mark the page as reusable
+     * rolling it as much as necessary.
+     * As above, the x86 h/w worked alright without this change
+     * however now large allocations work better and more reliably
+     * with on ARM, not to mention better RSS number accuracy
+     * for tools based on task_info API.
+     */
+    static void notify_not_using(void* p, size_t size) noexcept
+    {
+      SNMALLOC_ASSERT(is_aligned_block<PALBSD::page_size>(p, size));
+      while (madvise(p, size, MADV_FREE_REUSABLE) == -1 && errno == EAGAIN)
+        ;
+    }
+
+    /**
+     * same remark as above but we need to mark the page as REUSE
+     * first
+     */
+    template<ZeroMem zero_mem>
+    static void notify_using(void* p, size_t size) noexcept
+    {
+      SNMALLOC_ASSERT(
+        is_aligned_block<PALBSD::page_size>(p, size) || (zero_mem == NoZero));
+      os_unfair_lock lck = OS_UNFAIR_LOCK_INIT;
```

Comment:
Not sure if there are some crazy macro tricks here, but should it be static?
```suggestion
      static os_unfair_lock lck = OS_UNFAIR_LOCK_INIT;
```

Reply:
true